crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -1678,6 +1695,14 @@ impl<T: PoolTransaction> AllTransactions<T> {
             })
         }
 
+        if transaction.priority_fee_or_price() < self.minimal_priority_fee as u128 {
+            return Err(InsertErr::PriorityFeeBelowMinimum {
+                priority_fee: transaction.priority_fee_or_price(),
+                minimal_priority_fee: self.minimal_priority_fee as u128,
+                transaction: Arc::new(transaction),
+            })
```

Comment:
we currently enforce this check here during the validation part:

https://github.com/paradigmxyz/reth/blob/60940dd243918a1ba55f5cce5046a443bc75611d/crates/transaction-pool/src/validate/eth.rs#L439-L449

which seems more appropriate

although this only applies this check to dynamic fee txs because for legacy txs this setting would be dynamic

Reply:
Great, updated code to reflect that. Also don't think we need to handle legacy txs as they don't have a priority fee concept.